docs/spec.md

```diff
@@ -855,30 +857,31 @@ Afterwards, within each `process_group`:
 ![](images/spec/all_to_all.svg)
 
 Within each process group in the StableHLO process grid, splits the values of
-the `operand` tensor along `split_dimension` into parts, scatters the split
+the `operands` tensors along `split_dimension` into parts, scatters the split
 parts between the processes, concatenates the scattered parts along
-`concat_dimension` and produces a `result` tensor.
+`concat_dimension` and produces a `results` tensors. When multiple `operands`
+are specified, the `all_to_all` is performed on each operand.
 
 The operation splits the StableHLO process grid into `process_groups` which is
 defined as follows:
 
 * `cross_replica(replica_groups)` if `channel_id <= 0`.
 * `cross_partition(replica_groups)` if `channel_id > 0`.
 
-Afterwards, within each `process_group`:
+Afterwards, within each `process_group` and for each i in [0, size(operands)]:
```

Comment:
* `split_parts...@sender = split(operands...@sender, split_count, split_dimension)`
  for all `sender` in `process_group`.
* `scattered_parts...@receiver = [split_parts...@sender[receiver_index] for
  sender in process_group]` where
  `receiver_index = process_group.index(receiver)`.
* `results...@process = concatenate(scattered_parts...@process, concat_dimension)`.

Reply:
thank again for the spec review